src/Symfony/Component/Form/Form.php

```diff
@@ -562,6 +562,7 @@ public function submit($submittedData, $clearMissing = true)
 
                 foreach ($this->children as $name => $child) {
                     $isSubmitted = array_key_exists($name, $submittedData);
+                    $clearMissing = $this->getConfig()->getOption('expanded') ?: $clearMissing;
```

Comment:
@fabpot I have the same feeling, but I miss to find another way... expanded choice type seems to be a really singular case.

Feel free to close this PR if there is a better fix.